healthcareai/common/model_eval.py

```diff
@@ -37,140 +51,122 @@ def clfreport(modeltype, debug, devcheck, algo, X_train, y_train, X_test, y_test
 
-        if modeltype == 'classification':
-            print('\nMetrics:')
-            roc_auc = roc_auc_score(y_test, y_pred)
-            print('AU_ROC ScoreX:', roc_auc)
-
-            precision, recall, thresholds = precision_recall_curve(y_test, y_pred)
-            pr_auc = auc(recall, precision)
-            print('\nAU_PR Score:', pr_auc)
-
-        elif modeltype == 'regression':
-            print('##########################################################')
-            print('Model accuracy:')
-            print('\nRMSE error:', math.sqrt(mean_squared_error(y_test,
-                                                                y_pred_class)))
-            print('\nMean absolute error:', mean_absolute_error(y_test,
-                                                                y_pred), '\n')
-            print('##########################################################')
 
         # TODO: refactor this logic to be simpler
-        # Return without printing variable importance for linear case
-        if (not hasattr(clf, 'feature_importances_')) and (not
-            hasattr(clf, 'best_estimator_')):
+        # These returns are TIGHTLY coupled with their uses in develop and deploy. Both will have to be unwound together
+        has_importances = hasattr(clf, 'feature_importances_')
+        has_best_estimator = hasattr(clf, 'best_estimator_')
 
+        if not has_importances and not has_best_estimator:
+            # Return without printing variable importance for linear case
             return y_pred, roc_auc
-
-        # Print variable importance if rf and not tuning
-        elif hasattr(clf, 'feature_importances_'):
-            write_feature_importances(clf.feature_importances_,
-                                          col_list)
-
+        elif has_importances:
+            # Print variable importance if rf and not tuning
+            write_feature_importances(clf.feature_importances_, col_list)
             return y_pred, roc_auc, clf
-
-        # Print variable importance if rf and tuning
         elif hasattr(clf.best_estimator_, 'feature_importances_'):
-            write_feature_importances(
-                clf.best_estimator_.feature_importances_,
-                col_list)
-
+            # Print variable importance if rf and tuning
+            write_feature_importances(clf.best_estimator_.feature_importances_, col_list)
             return y_pred, roc_auc, clf
 
-    elif devcheck == 'notdev':
+    elif develop_model_mode is False:
+        y_pred = do_deploy_mode_stuff(X_test, X_train, clf, debug, model_type, use_saved_model, y_pred, y_train)
 
-        clf = algo
+    # TODO is it possible to get to this return if you are in develop_model_mode?
+    return y_pred
 
-        if use_saved_model is True:
 
-            clf = joblib.load('probability.pkl')
+def do_deploy_mode_stuff(X_test, X_train, clf, debug, mode_ltype, use_saved_model, y_pred, y_train):
```

Comment:
Typo in `mode_ltype` var name.

Could `clf` name be named to somehow represent the fact that an algorithm (and instantiated class) is being passed? `instantiated_algo_clf` ?